internal/isoeditor/rhcos.go

```diff
@@ -430,3 +436,71 @@ func writeOffsetInfo(headerOffset int64, offsetInfo *OffsetInfo, iso *os.File) (
 
 	return bytesLength, nil
 }
+
+// GetEmbedArea takes the 24 bytes of a CoreOS ISO and validates it is a
+// CoreOS ISO file before returning the offset into the file (and length) of
+// the embed area.
+func GetEmbedArea(header []byte) (int64, int64, error) {
```

Comment:
we could reuse readHeader func once merged: https://github.com/openshift/assisted-service/pull/1151/files#diff-f6c1ca589f46921957fccf7717574392fd4c4c84936f3cf74fcc59c16e15c866R407

Reply:
alternatively, `readHeader` could just call `GetEmbedArea` with the relevant header bytes and get the offset, length. :sunglasses:

Reply:
readHeader just reads the entire 24bit chunk and writes it into OffsetInfo struct (using binary.Read), so no need to actually parse the bytes manually.